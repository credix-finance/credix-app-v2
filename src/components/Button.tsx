import React from "react";
import { Button as AntdButton } from "antd";
import { ButtonProps as AntButtonProps } from "antd/lib/button";

interface ButtonProps {
	/**
	 * Controls whether the button is disabled or not.
	 */
	disabled?: AntButtonProps["disabled"];
	/**
	 * Optional icon that is renderd at the start of the button.
	 */
	icon?: AntButtonProps["icon"];
	/**
	 * Controls whether a loading spinner is displayed at the start of the button.
	 */
	loading?: AntButtonProps["loading"];
	/**
	 * Buttons with the block property will take up the entire width of their parent.
	 */
	block?: AntButtonProps["block"];
	onClick?: AntButtonProps["onClick"];
	className?: AntButtonProps["className"];
	children?: AntButtonProps["children"];
	/**
	 * The type of the button. Only "default" and "primary" are supported.
	 */
	type?: "default" | "primary";
	/**
	 * The html type of the button.
	 */
	htmlType?: AntButtonProps["htmlType"];
}

export const buttonTypeStyles = {
	primary: `
				bg-action-primary border-action-primary text-credix-primary
				hover:bg-neutral-60 hover:border-neutral-60
				active:bg-neutral-40 active:border-neutral-40 active:text-neutral-100
				disabled:bg-action-disable disabled:border-transparent disabled:text-disabled
			`,
	default: `
				bg-credix-primary text-neutral-100 border-neutral-100
				hover:text-neutral-60 hover:border-neutral-60
				active:text-neutral-40 active:border-neutral-60
				disabled:border-action-disable disabled:text-disabled
			`,
};

export const defaultButtonStyles =
	"rounded-md h-12 text-base font-medium border text-shadow-none shadow-none flex items-center justify-center gap-2 px-6";

export const Button = ({ children, className, type = "primary", ...props }: ButtonProps) => {
	className = [defaultButtonStyles, buttonTypeStyles[type], className].filter(Boolean).join(" ");

	return (
		<AntdButton className={className} type={type} {...props}>
			{children}
		</AntdButton>
	);
};
