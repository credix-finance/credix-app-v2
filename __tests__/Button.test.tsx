import React from "react";
import renderer from "react-test-renderer";
import { Button } from "@components/Button";

test("Primary button", () => {
	const component = renderer.create(<Button type="primary">Button</Button>);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
