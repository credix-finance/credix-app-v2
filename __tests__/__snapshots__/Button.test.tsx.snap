// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primary button 1`] = `
<button
  className="ant-btn ant-btn-primary rounded-[1px] text-shadow-none shadow-none flex items-center gap-2 px-[25px] 
				bg-action-primary border-action-primary text-credix-primary
				hover:bg-neutral-60 hover:border-neutral-60
				active:bg-neutral-40 active:border-neutral-40 active:text-neutral-100
				disabled:bg-action-disable disabled:border-transparent disabled:text-disabled
			 text-base font-medium h-[42px] "
  onClick={[Function]}
  type="button"
>
  <span>
    Button
  </span>
</button>
`;
